### ISSUE TITLE
Content below navigation does not have a top margin

### ISSUE THREAD

**Issue by ralphboeije** [part-up/part-up#1183]
On the profile page: the content below the user navigation does not have a top margin and is shown directly under the navigation.

**Comment by ralphboeije** [comment]
As you can see on the screenshot below (taken on iPad), the orange buttons are behind the navigation (drag the videos down and you see what I mean). The title of the page as well.

![7492f258-ce18-48b9-b803-0377a97f0136](https://user-images.githubusercontent.com/14044235/33400297-e38c08dc-d555-11e7-8a4a-4623b28f10da.png)

**Comment by jyoopensource** [comment]
Is that orange button is chat? And is it appearing correctly on the desktop?

**Comment by ralphboeije** [comment]
@jyoopensource no, that is a button to add video and photo, but that only shows on your own profile of course (sorry, should have mentioned that). If you pull/scroll down, you also see the content title on the left appear from under the “Who is Ralph”, etc navigation elements.

**Comment by jyoopensource** [comment]
ok...Is it possible to provide the expected screenshot?
Also i am new to this and need some help on where those css files are located and little hand holding to get familiarized.

**Comment by ralphboeije** [comment]
@noelheesen could you give @jyoopensource a hand to know where to look?

I don't have a design screenshot for you, but if you scroll up your own profile (found behind your picture in the top right corner), then you see the buttons in full. See screenshot below. 

![schermafbeelding 2017-11-30 om 09 39 22](https://user-images.githubusercontent.com/14044235/33421508-92966a6e-d5b2-11e7-98ba-a7d26ab6657a.png)

**Comment by noelheesen** [comment]
Yes, `app/packages/partup-client-pages/profile/profile.html`. I see that when I navigate to my own profile I don't have the margin issue, very interesting.

![profileexample](https://user-images.githubusercontent.com/18031130/33431020-08526a3a-d5d3-11e7-9110-4744eeadd65e.PNG)

**Comment by ralphboeije** [comment]
Strange indeed @noelheesen I have reproduced it on FF, Safari and Chrome on Mac.

**Comment by jpiduru** [comment]
@ralphboeije, @noelheesen  - Would you guys like to give another issue for me?

**Comment by ralphboeije** [comment]
@jyoopensource Why can't you fix this one? It is a reproducable bug and Noel has lead you the way where you can find the place to fix it.

**Comment by jyoopensource** [comment]
Okay , i'll look into it.

**Comment by jpiduru** [comment]
![image](https://user-images.githubusercontent.com/19610897/33490082-865cae1a-d67b-11e7-8b3a-06f15ce4802d.png)

Chrome screen shot

![image](https://user-images.githubusercontent.com/19610897/33490169-d5ccd092-d67b-11e7-87cf-f9f06f2fb649.png)

**Comment by jpiduru** [comment]
@noelheesen -

 <div class="pu-wrap pu-wrap-columnlayout">
Where are the the .css files located to find the details of "pu-wrap pu-wrap-columnlayout" / or the css class details?

Thanks!

**Comment by jyoopensource** [comment]
@noelheesen - I mean , Can you point me the appropriate css files related profile.html? 

Thanks!

**Comment by noelheesen** [comment]
Hi, I'm sorry for the late response. I can't reproduce either. I think it may be related to having content like Ralph does. If you go to his [profile](https://part-up.com/profile/pbnjsmy75oDNP3vok) and scroll down you'll see his name is below the profile description and appears as soon as the description collapses.

If you inspect the profile header and scroll the header get's an extra class named `pu-sub-pageheader-layered-shrinked`. We use [sass](http://sass-lang.com/) for our css, this might be a bit confusing at first but is really cool when you get used to it. The sass for this issue is located in `app/client/stylesheets/components/_pu-page.scss`.

**Comment by ralphboeije** [comment]
indeed @noelheesen seems only to be an issue when there is content on this profile page, such as strength test results and/or images/videos. @jyoopensource that narrows the issue to a manageable level I guess?

**Comment by jpiduru** [comment]
@ralphboeije - Sure , i'll try...

**Comment by jpiduru** [comment]
Thanks @noelheesen . I'll go thru the .css and will get back with any questions.Appreciate your help!

**Comment by jyoopensource** [comment]
@ralphboeije ,@noelheesen , 

I tried few things -

1.First added videos to my profile to reproduce the error and am not able to reproduce it.
Attached the screen shot.
<img width="1631" alt="screen shot 2017-12-05 at 11 59 43 am" src="https://user-images.githubusercontent.com/34107803/33623295-24fbecd8-d9b6-11e7-8008-2377a42de69b.png">

2. Tried to login to part up dev environment using the prod credentials and my user is not found and i tried to sign up and got internal server error. 
<img width="1647" alt="screen shot 2017-12-05 at 12 09 42 pm" src="https://user-images.githubusercontent.com/34107803/33623371-5b5b898c-d9b6-11e7-85eb-d0c1b217ac96.png">

Am really not sure what else to do at this point. Any pointers to move forward is appreciated!

Thanks!

**Comment by ralphboeije** [comment]
HI @jyoopensource point 2: on dev (local host and staging.part-up.com) you can only log in with the fixture user accounts (please read https://github.com/part-up/part-up). WRT point 1, the only difference I see is that you did not take the strength test as I did. But when you visit: https://part-up.com/profile/pbnjsmy75oDNP3vok than you don't even see the results of that strength test (bug) and still the title (who is Ralph) is hidden behind the horizontal navigation.

**Comment by noelheesen** [comment]
You can login locally with `admin@example.com` pw:`user`. When you inspect the profile wrapper you can see a large padding to compensate for the profile header. The padding lines up with where the content starts

![profile-padding](https://user-images.githubusercontent.com/18031130/33736896-a32f75c2-db94-11e7-9c69-7d7962e0684f.png)

It's due to the profile height that it no longer lines up on some profiles, like that of Ralph.

![profile-wrong-padding](https://user-images.githubusercontent.com/18031130/33736898-a34eab40-db94-11e7-8f99-d7d50363a850.png)

The dynamic height of the profile header makes it hard to give a margin as it's not always needed.
For it to always align properly we need to give the profile header `pu-sub-pageheader` a fixed height. Ralphs header takes 312px, so I think 325px is safe to try. Then you can change the padding of `.pu-page-profile` accordingly. You can find the padding on line `303`. I don't know if it will look good on profiles without much information, you'll have to play around a bit.

Yours,

**Comment by jyoopensource** [comment]
Thanks @noelheesen.

I have updated the padding on line 303 to 325 px to start with for user john as that profile has some videos. And updated the browser and still nothing changed on UI.


<img width="1636" alt="screen shot 2017-12-07 at 5 09 18 pm" src="https://user-images.githubusercontent.com/34107803/33743243-9ffa1f7e-db71-11e7-9dc6-903cc5bff850.png">

2. Also , started checking the dev tools to see the changes per your suggestion .

I see below line 
<header class="pu-sub-pageheader pu-sub-pageheader-layered"> 
and NOT  pu-sub-pageheader-profile as my update happened for that class on 303, before/after my update. Just curious to know why that did appear. My thought process was to look at css value before and after the update.

screen shot for point 2.
<img width="1651" alt="screen shot 2017-12-07 at 5 13 27 pm" src="https://user-images.githubusercontent.com/34107803/33743496-a13e7c08-db72-11e7-89aa-f6e35db7cd51.png">

**Comment by noelheesen** [comment]
Hi @jyoopensource thanks for the patience. Depending on the editor you use you might have visual git control. If you have changes you can undo them there or via the command-line by running `git checkout -f`.

After further investigation I found the following:

`_pu-page.scss`:
line 233: `position: fixed` causes the header to not be calculated when rendering the layout. This is 'solved' by adding padding to the next element that get's rendered at line 302-304. This was fine once but we now require a profile-header that can vary in size, especially with the expand [fix](https://github.com/part-up/part-up/commit/355f2fe72abaa3b78327f4e5b5b98d1a523f2566).

If you remove line `233`, `234` and the block `302-304` you should see the 'Who is [name]' again.

Now something else breaks when scrolling down. The header collapses and only show the menu. This will set the scroll position differently and will trigger an infinite loop of expanding-collapsing.

You can find the cause in `app/packages/partup-client-pages/app/profile.js` on line `118`
```
shrinkHeader: function() {
  return Partup.client.scroll.pos.get() > 100;
},
```
Can you try to play around with the value? maybe a fixed value of 25 will be fine. Another option would be to calculate it. e.g.
```
const innerHeight = window.innerHeight;
innerHeight - (innerHeight / 10)
```
Just play around with it and test what feels right.

If you have any questions feel free to ask :)

Yours,

**Comment by Krijger** [comment]
To reproduce this issue all you need to do is edit your profile by adding a load of text in the "Introduce yourself" field. The styling bug is still there

**Comment by jyoopensource** [comment]
@noelheesen - Thanks Noel. sorry was out on vacation. I'll start looking into this issue again.
